src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -77,13 +81,13 @@ public static function create($data = null, $status = 200, $headers = [])
      *     return JsonResponse::fromJsonString('{"key": "value"}')
      *         ->setSharedMaxAge(300);
      *
-     * @param string|null $data    The JSON response string
-     * @param int         $status  The response status code
-     * @param array       $headers An array of response headers
+     * @param string $data    The JSON response string
+     * @param int    $status  The response status code
+     * @param array  $headers An array of response headers
      *
      * @return static
      */
-    public static function fromJsonString($data = null, $status = 200, $headers = [])
+    public static function fromJsonString($data, $status = 200, $headers = [])
```

Comment:
We cannot remove a default value. That is a [BC break](https://symfony.com/doc/current/contributing/code/bc.html#changing-classes)

Reply:
Maybe it is better to cast $data to a string on line 92?

Reply:
@Nyholm Calling that method without any arguments will 100% lead to a `TypeError`. I have proposed to remove the default value because `$data` effectively has always been mandatory.

Reply:
https://github.com/symfony/symfony/pull/39271#issuecomment-737516925

Reply:
> Calling that method without any arguments will 100% lead to a TypeError

```php
return new static((string) $data, $status, $headers, true);
```

This will work..

-------

But you mean, since calling `JsonResponse::fromJsonString()` has always generated an error, then it is safe to do this technical BC break?

Reply:
Oh, Im sorry. 

This is weird.. When I first looked at this PR I didnt see any comments or discussion except for @derrabus approval. Im not sure if I should blame Github or my own "word blindness"...


Forget what I said, I'll read the discussions now.